### PR TITLE
Deprecation removal of Gradle UI

### PR DESCRIPTION
**Pull request by lacasseio**
### Context
See https://github.com/gradle/gradle/issues/995

### Gradle Core Team Checklist
- [x] Verify [test coverage and CI build status](https://builds.gradle.org/project.html?projectId=Gradle_Branches_Checkpoints&tab=projectOverview&branch_Gradle_Branches_Checkpoints=dl-remove-gradle-ui-deprecated)
- [x] Mention in release notes

### CODE REVIEW

File: subprojects/docs/src/docs/release/notes.md [hunk 1]
```diff
@@ -192,6 +192,7 @@ The following are the newly deprecated items in this Gradle release. If you have
 - Moved classes `RhinoWorkerHandleFactory` and `RhinoWorkerUtils` into internal package.
 - Removed `RhinoWorker`.
 - Removed constructor `EarPluginConvention(Instantiator)`.
+- Removed the Gradle UI module. All classes for this feature have been removed as well as all leftovers supporting class from the Open API partly removed due to deprecation in Gradle 2.0.
```
Comment (bmuschko) [inline subprojects/docs/src/docs/release/notes.md]: I believe users rather know the UI module as the [Gradle GUI](https://docs.gradle.org/current/userguide/tutorial_gradle_gui.html).

File: subprojects/launcher/launcher.gradle [hunk 2]
```diff
@@ -12,11 +12,11 @@ dependencies {
     compile project(':jvmServices')
     compile project(':core')
     compile project(':cli')
-    compile project(':ui')
     compile project(':toolingApi')
     compile project(':native')
     compile project(':logging')
 
+    compile libraries.commons_io
```
Comment (bmuschko) [inline subprojects/launcher/launcher.gradle]: Why was that necessary?

Comment (lacasseio) [inline subprojects/launcher/launcher.gradle]: `libraries.commons_io` was coming as [a transitive dependency of `compile project(':ui')`](https://github.com/gradle/gradle/pull/1749/files#diff-d527a28ac411deca36c287fe8b663190L26). To allow `:launcher` project to compile, this was required. We can look at the code to remove this dependency if you think it shouldn't be required.

File: subprojects/docs/src/docs/release/notes.md (hunk 1 above)
Comment (lacasseio) [inline subprojects/docs/src/docs/release/notes.md]: Done

File: subprojects/docs/src/docs/release/notes.md [hunk 3]
```diff
@@ -192,7 +192,7 @@ The following are the newly deprecated items in this Gradle release. If you have
 - Moved classes `RhinoWorkerHandleFactory` and `RhinoWorkerUtils` into internal package.
 - Removed `RhinoWorker`.
 - Removed constructor `EarPluginConvention(Instantiator)`.
-- Removed the Gradle UI module. All classes for this feature have been removed as well as all leftovers supporting class from the Open API partly removed due to deprecation in Gradle 2.0.
+- Removed the [Gradle GUI](https://docs.gradle.org/3.5/userguide/tutorial_gradle_gui.html) module. All classes for this feature have been removed as well as all leftovers supporting class from the Open API partly removed due to deprecation in Gradle 2.0.
```
Comment (bmuschko) [inline subprojects/docs/src/docs/release/notes.md]: I'd just say "Gradle GUI" without module.

Comment (lacasseio) [inline subprojects/docs/src/docs/release/notes.md]: My bad for not understanding correctly the first time, done.

**Comment by lacasseio** [comment]
Your requested change were done @bmuschko.
